clang/include/clang/Basic/BuiltinsAMDGPU.def

```diff
@@ -149,6 +149,19 @@ BUILTIN(__builtin_amdgcn_mqsad_pk_u16_u8, "WUiWUiUiWUi", "nc")
 BUILTIN(__builtin_amdgcn_mqsad_u32_u8, "V4UiWUiUiV4Ui", "nc")
 
 BUILTIN(__builtin_amdgcn_make_buffer_rsrc, "Qbv*sii", "nc")
+BUILTIN(__builtin_amdgcn_raw_ptr_buffer_store_i8, "vcQbiiIi", "n")
```

Comment:
Probably should drop the _ptr part of the name. This was more of a legacy issue in the intrinsic case, since the resource-as-vector case already took the name.

Also not sure if we should follow the naming convention of the instruction instead (probably the gfx12 one?).

Also should document these somewhere

Reply:
> Probably should drop the _ptr part of the name. This was more of a legacy issue in the intrinsic case, since the resource-as-vector case already took the name.

Sure, can do.

> Also not sure if we should follow the naming convention of the instruction instead (probably the gfx12 one?).

you mean things like `buffer_store_b8`? In this way, we don't need to differentiate types, just bit size.

Reply:
Yes. Pre-gfx12 it was _byte, _short, _dword, _dwordxN. But that also means you need casts and we drop the FP and 16-element versions so I don't know which way is best. Ideally we would just use the elementwise intrinsic approach but that won't work for loads

Reply:
Plus we can always write the instcombine to fold the bitcast into the intrinsic signature